test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs

```diff
@@ -2931,6 +2938,323 @@ public void PackageBuilderPreserveFileLastWriteTime()
+                                    numberOfDateNotCorrectedFiles++;
+                                }
+                                else
+                                {
+                                    // File from 12/31/2107, 23:59:58 UTC and after that
+                                    Assert.True(entry.LastWriteTime.DateTime >= ZipFormatMaxDate.AddSeconds(-1) && entry.LastWriteTime.DateTime <= ZipFormatMaxDate);
+                                    numberOfDateCorrectedFiles++;
+                                }
+                            }
+                        }
+                    }
+                }
+
+                Assert.True(numberOfDateNotCorrectedFiles == 2);
+                Assert.True(numberOfDateCorrectedFiles == 5);
+                Assert.Equal(innerLogger.LogMessages.Count, 1);
+                Assert.True(innerLogger.LogMessages.First().Message.Contains("because the zip file format does not support timestamp values before"));
+            }
+        }
+
+        [Fact]
+        public async Task PackAndExtract_TestLastWriteTimeForZipfile_UsingLocalTimeAsync()
+        {
+            // https://github.com/NuGet/Home/issues/7395
+            // Implicitly testing timezone issue (Our case testing on PST timezone).
+
+            // Arrange
+            var now = DateTime.Now;
+            TestLogger innerLogger = new TestLogger();
+            ILogger logger = new PackCollectorLogger(innerLogger, new WarningProperties());
+
+            using (var root = TestDirectory.Create())
+            {
+                var resolver = new PackagePathResolver(root);
+                var packageName = "A";
+                var outputNupkg = Path.Combine(root, $"{packageName}.nupkg");
+                var packageA = new PackageIdentity(packageName, new NuGetVersion("2.0.3"));
+                var builder = new PackageBuilder(logger) { Id = packageA.Id, Version = packageA.Version, Description = "test description" };
+                builder.Authors.Add("Test author");
+
+                var fileList = new List<string>()
+                {
+                    Path.Combine(@"build","foo.props"),
+                    Path.Combine(@"buildCrossTargeting","foo.props"),
+                    Path.Combine(@"buildMultiTargeting","foo.props"),
+                    Path.Combine(@"buildTransitive", "foo.props"),
+                    Path.Combine(@"buildTransitive", "net5.0", "foo.props"),
+                    Path.Combine(@"content", "foo.jpg"),
+                    Path.Combine(@"contentFiles", "any", "any", "foo.png"),
+                    Path.Combine(@"contentFiles", "cs", "net5.0", "foo.cs"),
+                    Path.Combine(@"embed", "net5.0", "foo.dll"),
+                    Path.Combine(@"lib", "net5.0", "foo.dll"),
+                    Path.Combine(@"ref", "net5.0", "foo.dll"),
+                    Path.Combine(@"runtimes", "win", "native", "foo.o"),
+                    Path.Combine(@"tools", "foo.dll"),
```

Comment:
Since all files have the same assertion, is there any benefit to having more than 1 file?